sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/ApiExtensions.kt

```diff
@@ -390,6 +390,43 @@ fun KtNameReferenceExpression.findPreviousUsages(
             }
     }
 
+/**
+ * Will try to find all usages of this reference. You can provide a searchStartNode to define the scope of the search (will
+ * use the closest block statement going up the AST). You can also provide a predicate to filter findings.
+ */
+fun KtNameReferenceExpression.findAllUsages(
+    searchStartNode: KtExpression = this,
+    predicate: (KtNameReferenceExpression) -> Boolean = { _ -> true }
+) =
+    mutableListOf<KtNameReferenceExpression>().also { acc ->
+        searchStartNode.getParentOfType<KtBlockExpression>(false)
+            ?.collectDescendantsOfType<KtNameReferenceExpression> { it.getReferencedName() == this.getReferencedName() }
+            ?.let { usages ->
+                for (usage in usages) {
+                    if (usage === this) continue
+                    else if (predicate(usage)) {
+                        acc.add(usage)
+                    }
+                }
+            }
+    }
+
+fun KtProperty.usages(
+    searchStartNode: KtExpression = this,
+    predicate: (KtNameReferenceExpression) -> Boolean = { _ -> true }
+) =
+    mutableListOf<KtNameReferenceExpression>().also { acc ->
+        searchStartNode.getParentOfType<KtBlockExpression>(false)
+            ?.collectDescendantsOfType<KtNameReferenceExpression> { it.getReferencedName() == name }
+            ?.let { usages ->
+                for (usage in usages) {
+                    if (predicate(usage)) {
+                        acc.add(usage)
+                    }
+                }
+            }
+    }
```

Comment:
These and the `findPreviousUsages` fun look super similar - can we generalize them?